rotkehlchen/exchanges/htx.py

```diff
@@ -188,13 +204,135 @@ def query_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
+        started to use the api.
+        """
+        log.debug(f'querying HTX online with {start_ts=}-{end_ts=}')
+        raw_data = self._paginated_query(
+            endpoint='/v1/query/deposit-withdraw',
+            options={
+                'type': 'deposit' if query_for == AssetMovementCategory.DEPOSIT else 'withdraw',
+                'size': PAGINATION_LIMIT,
+            },
+        )
+        movements = []
+        for movement in raw_data:
+            if (timestamp_raw := movement.get('updated-at')) is not None:
+                timestamp = ts_ms_to_sec(TimestampMS(int(timestamp_raw)))
+                if not start_ts <= timestamp <= end_ts:
+                    continue  # skip if is not in range
+            else:
+                continue  # skip missing timestamp
+
+            try:
+                coin = asset_from_htx(movement['currency'])
+            except UnknownAsset:
+                log.error(f'Could not read assets from trade pair {movement["currency"]}')
+                continue
+
+            try:
+                movements.append(AssetMovement(
+                    timestamp=timestamp,
+                    location=Location.HTX,
+                    category=AssetMovementCategory.DEPOSIT if movement['type'] == 'deposit' else AssetMovementCategory.WITHDRAWAL,  # noqa: E501
+                    address=movement['address'],
+                    transaction_id=movement['tx-hash'],
+                    asset=coin,
+                    amount=deserialize_asset_amount(movement['amount']),
+                    fee_asset=coin,
+                    fee=deserialize_fee(movement['fee']) if movement['fee'] != 0 else Fee(ZERO),
+                    link=str(movement['id']),
+                ))
+            except (DeserializationError, KeyError) as e:
+                msg = str(e)
+                if isinstance(e, KeyError):
+                    msg = f'Missing key {msg}'
+
+                log.error(f'{e} when reading movement for HTX deposit {movement}')
+                continue
+
+        return movements
+
     def query_online_deposits_withdrawals(
             self,
             start_ts: Timestamp,
             end_ts: Timestamp,
     ) -> list[AssetMovement]:
         """Query deposits and withdrawals sequentially"""
-        return []  # TODO: Implement this in another PR
+        new_movements = self._query_deposits_withdrawals(
+            start_ts=start_ts,
+            end_ts=end_ts,
+            query_for=AssetMovementCategory.DEPOSIT,
+        )
+        new_movements.extend(
+            self._query_deposits_withdrawals(
+                start_ts=start_ts,
+                end_ts=end_ts,
+                query_for=AssetMovementCategory.WITHDRAWAL,
+            ),
+        )
+        return new_movements
+
+    @staticmethod
+    def remove_fee_currency(symbol: str, fee_currency: str) -> str:
+        """
+        Removes the 'fee-currency' value from the 'symbol' if it appears at the beginning or
```

Comment:
don't put it in the class since if you have multple instances you copy it. Make it a function in this file